docs/DafnyRef/integration-java/IntegrationJava.md

```diff
@@ -0,0 +1,215 @@
+
+If the Dafny method (say `p()`) is not a member of any class and is in the top-level 
+(unnamed) module, then it is called as `_System.__default.p()`
+
+## **Specifying class path**
+
+The dafny build and run steps invoke `javac` and `java`. Dafny will use the version of those tools that would be invoked at a terminal prompt (e.g., the ones
+found by a Linux `$PATH`). 
+There is no mechanism to supply options to these internal uses of `javac` and `java`. However, `dafny` does use the value of the environment variable `CLASSPATH`. For example, if you are writing a Dafny program, `Demo.java`, which calls
+methods in `Demo.jar`, you can build and run the program using
+```CLASSPATH=`pwd`/Demo.jar dafny run -t:java Demo.dfy```.
+The contents of the CLASSPATH must all be absolute paths.
+
+There are two means to include other options in the build. 
+First, one could create scripts named `java` and `javac` that invoke the actual `java` and `javac` with the desired options or environment settings, and then
+place theses scripts on the system `$PATH`.
+
+Alternately, one can use `dafny translate` to create the `.java` translation of the Dafny program and then use regular Java build processes to combine the
+Dafny-generated Java code with other Java code.
+
+## **Matching Dafny and Java types**
+
+If the Java method has input arguments or an output value, then the Dafny declaration must use
+corresponding types in the Dafny declaration:
+
+|-------------------------------|-----------------------------|
+|  Dafny type                   |   Java type                 |
+|-------------------------------|-----------------------------|
+| bool                          | boolean                     |
+| int                           | java.math.BigInteger        |
+| int64                         | long                        |
+| int32                         | int                         |
+| int16                         | short                       |
+| int8                          | byte                        |
+| char                          | char                        |
+| bitvector                     | appropriate int-based type  |
+| ORDINAL                       | java.math.BigInteger        |
+| real                          | dafny.BigRational           |
+|                               | double                      |
+|                               | float                       |
+| string                        | dafny.DafnySequence<? extends Character>  |
+| JavaString                    | java.lang.String                        |
+| C, C? (for class, iterator C) | (class) C                   |
+| (trait) T                     | (interface) T                |
+| datatype, codatatype          | (class) C                   |
+| subset type                   | same as base type           |
+| tuple                         | dafny.Tuple_n_              |
+| array<T>                      | T'[]                        |
```

Comment:
```suggestion
| array<T>                      | T[]                        |
```
I get it that you want to say that T should also be translated, but you did not define T' before. Maybe you should? Or just leave the apostrophe out.

Reply:
Added the definition